pkg/envtest/server.go

```diff
@@ -87,11 +87,34 @@ type Environment struct {
 	// ControlPlane is the ControlPlane including the apiserver and etcd
 	ControlPlane integration.ControlPlane
 
-	// Config can be used to talk to the apiserver.  It's automatically
-	// populated if not set using the standard controller-runtime config
+	// Config can be used to talk to the apiserver (insecure endpoint).
+	// It's automatically populated if not set using the standard controller-runtime config
 	// loading.
 	Config *rest.Config
 
+	// SecureConfig can be used to talk to the apiserver (secure endpoint).
+	// It's automatically populated if not set using the standard controller-runtime config
+	// loading.  This just contains secure endpoint and tlsconfig (no authn info).
+	// To use this config, you have to configure kube-apiserver with some authn module(static token, basic auth, etc.)
+	// and set your authentication info to this config.  For example:
+	//
+	// // basic authn plugin case
+	// te := &envtest.Environment{
+	//   KubeAPIServerFlags: append(
+	//     envtest.DefaultKubeAPIServerFlags,
+	//     "--basic-auth-file=my-file", "--authorization-mode=RBAC",
```

Comment:
@everpeace what about extending envtest to generate a [`token-auth-file`](https://kubernetes.io/docs/reference/access-authn-authz/authentication/) with a token that is member of the `system:masters` group so this just works by default?

Also, a test would be great :upside_down_face: 

If you do not have time to pick that up, we can also merge this as-is and I will address my comments in a follow-up.

Reply:
@alvaroaleman Thank you for your review!!

> If you do not have time to pick that up, we can also merge this as-is and I will address my comments in a follow-up.

sorry...  I couldn't spare time to do this currently.  I'm very glad if we take this way 🙇 🙇

Reply:
@alvaroaleman As I understand it you suggestion is just to provide a static user in the token-auth-file? I think it would be better to have an option to add individual users (maybe in addition to a default) so that you can specify particular groups that the user may be a member of.

One example of where this may be useful is testing CSRs, where the API server uses the requesting parties credentials to become part of the CSR request. Typically here you'd want to be able to specify the exact groups that the user belongs to to be able to get the right groups plumbed into that CSR

Reply:
@JoelSpeed you can just create a SA and then use impersonation for that. @DirectXMan12 is working on an updated version of this afaik